cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/CassandraCradleStorage.java

```diff
@@ -400,22 +401,18 @@ protected CompletableFuture<Void> doStoreGroupedMessageBatchAsync(GroupedMessage
 	{
 		PageId pageId = page.getId();
 
-		CompletableFuture<Void> future =  messagesWorker.storeGroupedMessageBatch(batch, pageId);
-
 		if (settings.isStoreIndividualMessageSessions()) {
+			CompletableFuture<Void> future = messagesWorker.storeGroupedMessageBatch(batch, pageId, false);
 			for (MessageBatchToStore b : batch.getSessionMessageBatches()) {
 				future = future.thenComposeAsync((unused) -> messagesWorker.storeMessageBatch(b, pageId), composingService)
 						.thenComposeAsync((unused) -> messagesWorker.storeSession(b), composingService)
 						.thenComposeAsync((unused) -> messagesWorker.storePageSession(b, pageId), composingService)
 						.thenAccept(NOOP);
 			}
+			return future;
 		} else {
-			future = future.thenApplyAsync((unused) -> getDirectionToSessionAliases(batch), composingService)
-					.thenComposeAsync((sessions) -> storeSessions(pageId, sessions), composingService)
-					.thenComposeAsync((sessions) -> storePageSessions(batch, sessions), composingService)
-					.thenAccept(NOOP);
+			return messagesWorker.storeGroupedMessageBatch(batch, pageId, true);
```

Comment:
This code will never be executed because of the exception on line 389

Reply:
the line 389 is instruction of doStoreMessageBatchAsync method, current method is doStoreGroupedMessageBatchAsync

Reply:
Sorry, my bad here